libs/execution/src/lib/transforms/transform-executor.ts

```diff
@@ -124,8 +142,12 @@ export class TransformExecutor {
         newColumn.push(newValue);
       }
 
-      context.evaluationContext.deleteValueForReference(inputDetails.port.name);
-    });
+      for (const inputDetails of inputDetailsList) {
+        context.evaluationContext.deleteValueForReference(
+          inputDetails.port.name,
+        );
+      }
```

Comment:
Consider extracting the loops for setting and deleting values for variables into their own helper functions.